src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspace.cs

```diff
@@ -16,7 +17,7 @@ namespace Microsoft.VisualStudio.LanguageServices
     /// </summary>
     public abstract class VisualStudioWorkspace : Workspace
     {
-        private readonly BackgroundCompiler _backgroundCompiler;
+        private BackgroundCompiler _backgroundCompiler;
```

Comment:
This is not a blocking comment/suggestion, but in general I prefer to leave things readonly even in the disposable case. To me the readonlyness is such a strong guarantee for what invariants the rest of the code can assume that losing it for just this case isn't worth it.